src/executioners/SpectralExecutionerLinearElastic.C

```diff
@@ -0,0 +1,402 @@
+    return 0.0;
+}
+
+InputParameters
+SpectralExecutionerLinearElastic::validParams()
+{
+  InputParameters params = SpectralExecutionerBase::validParams();
+  params.addClassDescription("Executioner for spectral solve of diffusion equation");
+  params.addParam<Real>("time_step", 1.0, "Time step for ODE integration");
+  params.addParam<unsigned int>("number_iterations", 0, "Maximum number of iterations for solver");
+  params.addParam<Real>("young_modulus", 1.0, "First parameter for isotropic materials");
+  params.addParam<Real>("poisson_ratio", 1.0, "Second parameter for isotropic materials");
+  params.addParam<Real>("average_material_factor", 1.0, "Homogeneized factor for multiphase");
+  params.addRequiredParam<std::vector<Real>>(
+      "global_strain_tensor",
+      "Vector of values defining the constant applied global strain "
+      "to add. Components are XX, YY, ZZ, YZ, XZ, XY");
+  params.addParam<Real>("solver_error", 1.0e-4, "Error for fixed iteration solver");
+
+  return params;
+}
+
+SpectralExecutionerLinearElastic::SpectralExecutionerLinearElastic(
+    const InputParameters & parameters)
+  : SpectralExecutionerBase(parameters),
+    _dt(getParam<Real>("time_step")),
+    _nsteps(getParam<unsigned int>("number_iterations")),
+    _young_modulus(getParam<Real>("young_modulus")),
+    _poisson_ratio(getParam<Real>("poisson_ratio")),
+    _average_factor(getParam<Real>("average_material_factor")),
+    _solver_error(getParam<Real>("solver_error"))
+{
+
+  _initial_strain_tensor.fillFromInputVector(getParam<std::vector<Real>>("global_strain_tensor"));
+
+  _t_current = 0.0;
+}
+
+void
+SpectralExecutionerLinearElastic::fillOutEpsilonBuffer(
+    FFTBufferBase<RankTwoTensor> & epsilon_buffer)
+{
+  const auto & grid = epsilon_buffer.grid();
+  const int ni = grid[0];
+  const int nj = grid[1];
+  const int nk = grid[2];
+
+  std::size_t index = 0;
+
+  for (int freq_x = 0; freq_x < ni; ++freq_x)
+    for (int freq_y = 0; freq_y < nj; ++freq_y)
+      for (int freq_z = 0; freq_z < nk; ++freq_z)
+      {
+        epsilon_buffer.realSpace()[index] = _initial_strain_tensor;
+        index++;
+      }
+}
+
+void
+SpectralExecutionerLinearElastic::getGreensFunction(FFTBufferBase<RankFourTensor> & gamma_hat,
+                                                    const RankFourTensor & elasticity_tensor)
+{
+  const auto & grid = gamma_hat.grid();
+  const int ndim = 3;
+
+  const int ni = grid[0];
+  const int nj = grid[1];
+  const int nk = (grid[2] >> 1) + 1;
+
+  std::size_t index = 0;
+
+  const auto & ivec = gamma_hat.kTable(0);
+  const auto & jvec = gamma_hat.kTable(1);
+  const auto & kvec = gamma_hat.kTable(2);
+
+  const std::vector<int> grid_vector = gamma_hat.grid();
+
+  const Complex I(0.0, 1.0);
+
+  // Fill fourth-order Green operator based on homogeneous material properties
+  for (int freq_x = 0; freq_x < ni; ++freq_x)
+    for (int freq_y = 0; freq_y < nj; ++freq_y)
+      for (int freq_z = 0; freq_z < nk; ++freq_z)
```

Comment:
this would better be written as

```
for (auto ivec : gamma_hat.kTable(0))
  for (auto jvec : gamma_hat.kTable(1))
    for (auto kvec : gamma_hat.kTable(2))
```

it would save a bunch of lines (92-94) and e.g. `ivec[freq_x]` would just become `ivec`.